### PR TITLE
add lint rule 'import/no-extraneous-dependencies'

### PR DESCRIPTION
**Pull request by bengl**
### What does this PR do?
Ensures that any thing that is `require`d is in package.json.

### Motivation
It's generally bad to require modules that aren't in package.json.

### CODE REVIEW

**Review by Qard — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -73,6 +73,7 @@
     "path-to-regexp": "^0.1.2",
     "performance-now": "^2.1.0",
     "protobufjs": "^6.9.0",
+    "resolve": "^1.20.0",
```
Comment (rochdev) [inline package.json:76]: @bengl This is added as a dependency here but a devDependency in `LICENSE-3rdparty.csv`. Where is it actually used?

Comment (bengl) [reply package.json:76]: https://github.com/DataDog/dd-trace-js/blob/e3766edae089944859941285d110425185ff32ec/packages/datadog-plugin-cucumber/src/index.js#L2

Comment (rochdev) [reply package.json:76]: Then `LICENSE-3rdparty.csv` should be updated accordingly.

Comment (rochdev) [reply package.json:76]: Fixed in https://github.com/DataDog/dd-trace-js/pull/1457

**Pull request merged**